src/Accounting/index.ts

```diff
@@ -0,0 +1,202 @@
+ */
+export const calcMinimumMargin: (quote: number, base: number, price: number, maxLeverage: number) => number = (
+    quote,
+    base,
+    price,
+    maxLeverage,
+) => {
+    const margin = totalMargin(base, quote, price);
+    if (margin > 0 || base < 0) {
+        return LIQUIDATION_GAS_COST * RYAN_6 + calcNotionalValue(base, price) / maxLeverage;
+    } else {
+        return 0;
+    }
+};
+
+/**
+ * Total margin of an account given a position
+ *  aka equity, the amount owned by the user if they cashed out at the given price
+ * @param quote Amount of quote asset
+ * @param base Amount of base asset, this is considered the position of the account
+ * @param price The given price of the asset 
+ * @returns the total margin of an account
+ */
+export const totalMargin: (base: number, quote: number, price: number) => number = (base, quote, price) =>
+    quote + base * price;
+
+/**
+ * Calculates a theoretical market exposure if it took all the 'best' orders it could
+ *  Returns this exposure and the orders that allow it to gain this exposure
+ * @param quote Amount of quote asset
+ * @param leverage leverage of the trade this could be passed in as quote leverage * quote
+ */
+export const calcTradeExposure: (
+    quote: number,
+    leverage: number,
+    orders: FlatOrder[],
+) => { exposure: number, slippage: number, tradePrice: number } = (quote, leverage, orders) => {
+    if (orders.length) {
+        // weighted average of the price, where the weights are the amounts at each price
+        let exposure = 0,
+            sumOfWeights = 0,
+            totalUnits = 0;
+        let deposit = quote * leverage; // total units of underlying
+        for (const order of orders) {
+            const amount = order.amount;
+            const orderPrice = order.price;
+            // remainding units of accounts quote use
+            const r = deposit - amount * orderPrice;
+            if (r >= 0) { // if it can eat the whole order
+                totalUnits += orderPrice * amount;
+                sumOfWeights += amount;
+                exposure += amount * orderPrice; // units of the assets
+                deposit -= amount * orderPrice; // subtract the remainder in units of underLying
+            } else { // eat a bit of the order nom nom
+                // if we get here the max amount we can is the remainder of deposit
+                if (deposit) {
+                    totalUnits += deposit * orderPrice;
+                    sumOfWeights += deposit;
+                    exposure += deposit / orderPrice;
+                }
+                break;
+            }
+        }
+        
+        const expectedPrice = orders[0].price;
+        // this is a weighted average of the prices and how much was taken at each price
+        const tradePrice = totalUnits ? totalUnits / sumOfWeights: expectedPrice;
+        return {
+            exposure: parseFloat(exposure.toFixed(10)),
+            slippage: 1 - expectedPrice/tradePrice,
```

Comment:
it'll be different for long and short. for long it'll be (trade price/expected price) - 1. for short it'll be (expected/trade) -1